test-tests/shared/src/test/scala/zio/test/GenSpec.scala

```diff
@@ -62,18 +412,128 @@ object GenSpec extends ZIOBaseSpec {
           } yield assert(result.reverse.headOption, isSome(equalTo((m, 0)) || equalTo((0, n))))
         }
       }
-    )
+    ),
+    testM("fromIterable constructs deterministic generators") {
```

Comment:
I think we can keep this test but we need to change `zipWith` to `crossWith` and mark it as `scala2Only` due to #2232.